src/error_diffusion.jl

```diff
@@ -96,9 +99,9 @@ function colordither!(
             px = img[r, c]
             alg.clamp_error && (px = clamp01(px))
 
-            col = closest_color(px, cs; metric=metric) # round to closest color
-            out[r, c] = col # apply pixel to dither
-            err = px - col  # diffuse "error" to neighborhood in filter
+            colorindex = argmin(colordiff.(px, labcs; metric=metric)) # find closest color
+            out[r, c] = colorindex # apply pixel to dither, which is an IndirectArray
+            err = px - cs[colorindex]  # diffuse "error" to neighborhood in filter
 
             for dr in drs
                 for dc in dcs
```

Comment:
I noticed this only recently; for the innermost for loop, a performance practice is to separate the loops into two parts: 1) border and 2) inner area. The benefits are 1) we can remove unnecessary bounds check for the inner area, and thus 2) we can enable `@simd` for such simple operations.

See https://github.com/JuliaImages/Images.jl/blob/9796acc8bf1d56d9574c555656fdba362f84a988/src/algorithms.jl#L505-L537 and https://github.com/JuliaImages/ImageBase.jl/pull/22#issuecomment-943333620

Reply:
Tracked in #49, thanks!